cro-clib/src/address.rs

```diff
@@ -0,0 +1,84 @@
+use crate::types::{CroAddressPtr, CroResult};
+use std::ffi::CString;
+use std::ptr;
+/// print address information
+#[no_mangle]
+/// # Safety
+pub unsafe extern "C" fn cro_print_address(
+    address_ptr: CroAddressPtr,
+    buf: *mut u8,
+    buf_length: u32,
+) -> CroResult {
+    let address = address_ptr.as_mut().expect("get address");
+    let address_string = address.address.to_string();
+    if address_string.as_bytes().len() >= buf_length as usize {
+        return CroResult::fail();
+    }
+    ptr::write_bytes(buf, 0, buf_length as usize);
+    ptr::copy_nonoverlapping(
+        address_string.as_bytes().as_ptr(),
+        buf,
+        address_string.as_bytes().len(),
+    );
+    CroResult::success()
+}
+
+/// print address information
+/// 32 bytes
+#[no_mangle]
+/// # Safety
+pub unsafe extern "C" fn cro_export_private(address_ptr: CroAddressPtr, dst: *mut u8) -> CroResult {
+    let address = address_ptr.as_mut().expect("get address");
+    let src = address.privatekey.serialize();
+    assert!(32 == src.len());
+    ptr::copy_nonoverlapping(src.as_ptr(), dst, src.len());
+    CroResult::success()
+}
```

Comment:
this doesn't "print address information", it copies a serialised private key

Reply:
in c api, user handles privatekey for normal wallet.
unlike hdwallet, normal wallet is not deterministic.
so user manages privatekeys in c side.